### PR TITLE
added access bundle resource and support for access flow bundle target

### PR DESCRIPTION
**Pull request by tomerstein**

### CODE REVIEW

File: internal/services/access_targets.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		integration, diagnostics := convertIntegrationTargetTerraformModelToApi(integrationTarget, availableIntegrations.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		resultIntegrationTargets = append(resultIntegrationTargets, *integration)
+	}
+
+	return resultIntegrationTargets, nil
+}
```
Comment (tomerstein) [inline internal/services/access_targets.go:50]: moved logic from `services/access_flow.go`

File: internal/services/access_targets.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	}
+
+	if result == nil {
+		diagnostics := diag.Diagnostics{}
+		diagnostics.AddError("Client Error", fmt.Sprintf("Failed to get integration: (%s) with resource type (%s)", integrationTarget.Name.ValueString(), integrationTarget.ResourceType.ValueString()))
+		return nil, diagnostics
+	}
+
+	return result, nil
+}
```
Comment (tomerstein) [inline internal/services/access_targets.go:156]: moved logic from `services/access_flow.go`

File: internal/schemas/access_targets.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
+package schemas
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func GetIntegrationTargetSchema(required bool) schema.SetNestedAttribute {
```
Comment (tomerstein) [inline internal/schemas/access_targets.go:10]: This schema moved from `provider/resource_access_flow.go`

File: docs/resources/access_bundle.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
```
Comment (tomerstein) [inline docs/resources/access_bundle.md:2]: All the docs is autogenerated from the schemas

File: docs/resources/access_flow.md [hunk 5]
```diff
@@ -43,6 +43,11 @@ resource "apono_access_flow" "postgresql_prod" {
       permissions   = ["READ_ONLY", "READ_WRITE", "ADMIN"]
     }
   ]
+  bundle_targets = [
```
Comment (tomerstein) [inline docs/resources/access_flow.md:46]: All the docs is auto generated from the schemas

File: internal/services/access_targets.go (hunk 1 above)
Comment (DmitriyO) [reply internal/services/access_targets.go:50]: lets be doing such moves in separate PRs next time

File: internal/services/access_targets.go [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+		dataBundleTargets = append(dataBundleTargets, *bundle)
+	}
+
+	return dataBundleTargets, nil
+}
+
+func ConvertBundleTargetsTerraformModelToApi(ctx context.Context, aponoClient *apono.APIClient, bundleTargets []models.BundleTarget) ([]apono.AccessTargetBundleV1, diag.Diagnostics) {
+	availableBundles, _, err := aponoClient.AccessBundlesApi.ListAccessBundles(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "bundles", "")
```
Comment (DmitriyO) [inline internal/services/access_targets.go:74]: why here it's "list" "bundles" and other place (line 55) its "list" "access bundles"

File: internal/provider/resource_access_flow.go [hunk 7]
```diff
@@ -428,4 +380,11 @@ func (a *accessFlowResource) ValidateConfig(ctx context.Context, req resource.Va
 			"must be -1 or positive number",
 		)
 	}
+
+	if len(model.IntegrationTargets) == 0 && len(model.BundleTargets) == 0 {
```
Comment (DmitriyO) [inline internal/provider/resource_access_flow.go:384]: did we have validation for integration targets prior to this PR?

File: internal/provider/resource_access_flow.go [hunk 8]
```diff
@@ -45,33 +46,14 @@ func (a accessFlowResource) Schema(_ context.Context, _ resource.SchemaRequest,
...
-				MarkdownDescription: "Filter value",
-				Required:            true,
-			},
-		},
-	}
 	var identitySchema = schema.NestedAttributeObject{
 		Attributes: map[string]schema.Attribute{
 			"name": schema.StringAttribute{
-				MarkdownDescription: "Identity Name (in user type, use email address instead)",
+				MarkdownDescription: "Name of the grantee. When `type = context_attribute`, this is the shift name. When `type = group`, this is the group name. When `type = user`, this is the email address. **NOTE: If a non-unique name is used with 'group' type, Apono applies the access flow to all groups matching the name.**",
```
Comment (DmitriyO) [inline internal/provider/resource_access_flow.go:52]: check all copy with relevant people

File: internal/provider/resource_access_bundle_test.go [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func testAccAccessBundleResourceConfig(accessBundleName string) string {
+	return fmt.Sprintf(`
+provider apono {
+  endpoint = "http://api.apono.dev"
+  personal_token = "1234567890abcdefg"
+}
+
+resource "apono_access_bundle" "test_access_bundle_resource" {
```
Comment (DmitriyO) [inline internal/provider/resource_access_bundle_test.go:61]: is there better way to define such test strings than inlining

File: internal/services/access_bundle.go [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+	if len(diagnostics) > 0 {
+		return nil, diagnostics
+	}
+
+	data := apono.UpdateAccessBundleV1{
+		Name:               *apono.NewNullableString(&updateAccessBundleRequest.Name),
+		IntegrationTargets: updateAccessBundleRequest.IntegrationTargets,
+	}
+
+	return &data, nil
```
Comment (DmitriyO) [inline internal/services/access_bundle.go:52]: what is the diff between this 2?

**Review by DmitriyO — APPROVED** [review]

File: internal/provider/resource_access_flow.go (hunk 7 above)
Comment (tomerstein) [reply internal/provider/resource_access_flow.go:384]: before, the `integration_targets` field was required so basically yes

File: internal/services/access_bundle.go (hunk 10 above)
Comment (tomerstein) [reply internal/services/access_bundle.go:52]: first is the create Model and the seconds is the update model, I will update the first function name to be more clear

File: internal/services/access_targets.go (hunk 6 above)
Comment (tomerstein) [reply internal/services/access_targets.go:74]: mistake, will change to access bundles

**Pull request merged**